contracts/RealmDiamond/facets/ParcelRolesRegistryFacet.sol

```diff
@@ -0,0 +1,235 @@
+    /// @param _grantor The user that approved the operator.
+    /// @param _operator The user that can grant and revoke roles.
+    /// @return isApproved_ Whether the operator is approved or not.
+    function isRoleApprovedForAll(address _tokenAddress, address _grantor, address _operator) public view override returns (bool) {
+        return s.itemsRoleApprovals[_grantor][_tokenAddress][_operator];
+    }
+
+    /// @notice Checks whether an NFT has at least one non-revocable role.
+    /// @param _tokenAddress The token address.
+    /// @param _tokenId The token identifier.
+    /// @return true if the NFT is locked.
+    function _hasNonRevocableRole(address _tokenAddress, uint256 _tokenId) internal view returns (bool) {
+        for (uint256 i = 0; i < s.allowedRoles.length; i++) {
+            if (
+                !s.erc7432_roles[_tokenAddress][_tokenId][s.allowedRoles[i]].revocable &&
+                s.erc7432_roles[_tokenAddress][_tokenId][s.allowedRoles[i]].expirationDate > block.timestamp
+            ) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /** ERC-7432 View Functions **/
+
+    function ownerOf(address _tokenAddress, uint256 _tokenId) external view override returns (address owner_) {
+        return s.erc7432OriginalOwners[_tokenAddress][_tokenId];
+    }
+
+    function recipientOf(address _tokenAddress, uint256 _tokenId, bytes32 _roleId) external view override returns (address recipient_) {
+        if (s.erc7432_roles[_tokenAddress][_tokenId][_roleId].expirationDate > block.timestamp) {
+            return s.erc7432_roles[_tokenAddress][_tokenId][_roleId].recipient;
+        }
+        return address(0);
+    }
+
+    function roleData(address _tokenAddress, uint256 _tokenId, bytes32 _roleId) external view override returns (bytes memory data_) {
+        if (s.erc7432_roles[_tokenAddress][_tokenId][_roleId].expirationDate > block.timestamp) {
+            data_ = s.erc7432_roles[_tokenAddress][_tokenId][_roleId].data;
+        }
+        return data_;
+    }
+
+    function roleExpirationDate(address _tokenAddress, uint256 _tokenId, bytes32 _roleId) external view override returns (uint64 expirationDate_) {
+        if (s.erc7432_roles[_tokenAddress][_tokenId][_roleId].expirationDate > block.timestamp) {
+            return s.erc7432_roles[_tokenAddress][_tokenId][_roleId].expirationDate;
+        }
+        return 0;
+    }
+
+    function isRoleRevocable(address _tokenAddress, uint256 _tokenId, bytes32 _roleId) external view override returns (bool revocable_) {
+        return
+            s.erc7432_roles[_tokenAddress][_tokenId][_roleId].expirationDate > block.timestamp &&
+            s.erc7432_roles[_tokenAddress][_tokenId][_roleId].revocable;
+    }
+
+    /** ERC-165 Functions **/
+
+    function supportsInterface(bytes4 interfaceId) external view virtual override returns (bool) {
+        return interfaceId == type(IERC7432).interfaceId;
+    }
```

Comment:
Can we include this function on this facet? It might be implemented on another face of this contract. In fact, we store the `interfaceIds` on storage, so it should be reading from there